File: ruby/sieve/sieve.rb
class Sieve
  attr_accessor :primes
  def initialize(limit)
    @limit = limit
    @range = (2..limit)
    @composites = find_composites
    @primes = @range.to_a - @composites
  end

  def find_composites
    results = []
    @range.map do |n|
      @range.each do |x|
        break if x * n > @limit

        results.push(x * n)
      end
    end
    results
  end
end

File: javascript/sieve/sieve.js
// const range = (start, end) => new Array(end - start + 1).fill(undefined).map((value, index) => index + start);

function range(start, end) {
    if(start === end) return [start];
    return [start, ...range(start + 1, end)];
};

class Sieve {
  constructor(max) {
    this.max = max;
    this.range = range(2, max);
    console.log(this.range)
    this.composites = this.generateComposites();
    this.primes = this.range - this.composites
  }


  generateComposites() {

  }
}

// export default Sieve;

new Sieve(10)
